`./scripts/make gdal` errors in Ubuntu

Thanks for your work @ddohler !

Using the following OS:

`OS: Ubuntu 22.04.1 LTS x86_64`

`./scripts/make gdal` generates the following error:

```
make[3]: Entering directory '/opt/gdaljs/gdal/frmts/pcidsk'
/opt/emsdk/upstream/emscripten/em++ -c -DHAVE_LIBJPEG -I../jpeg/libjpeg -DPCIDSK_INTERNAL -Isdk  -I/opt/gdaljs/gdal/port  -DGDAL_COMPILATION -g -O2 -fPIC  -Wall -Wextra -Winit-self -Wunused-parameter -Wformat -Werror=format-security -Wno-format-nonliteral -Wshorten-64-to-32 -Wshadow -Werror=vla -Wdate-time -Wnull-dereference -Wextra-semi -Wcomma -Wfloat-conversion -Wdocumentation -Wno-documentation-deprecated-sync -Wunused-private-field -Wmissing-prototypes -Wmissing-declarations -Wnon-virtual-dtor -Woverloaded-virtual -fno-operator-names -Wzero-as-null-pointer-constant -Wsuggest-override -Wimplicit-fallthrough  sdk/core/clinksegment.cpp -o ../o/clinksegment.o
shared:INFO: (Emscripten: Running sanity checks)
sdk/core/clinksegment.cpp:79:27: error: no member named 'bind2nd' in namespace 'std'
                     std::bind2nd(std::not_equal_to<char>(), ' '));
                     ~~~~~^
1 error generated.
em++: error: '/opt/emsdk/upstream/bin/clang++ -target wasm32-unknown-emscripten -fignore-exceptions -fvisibility=default -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr -DEMSCRIPTEN -I/opt/emsdk/upstream/emscripten/cache/sysroot/include/SDL --sysroot=/opt/emsdk/upstream/emscripten/cache/sysroot -Xclang -iwithsysroot/include/compat -c -DHAVE_LIBJPEG -I../jpeg/libjpeg -DPCIDSK_INTERNAL -Isdk -I/opt/gdaljs/gdal/port -DGDAL_COMPILATION -g3 -O2 -fPIC -Wall -Wextra -Winit-self -Wunused-parameter -Wformat -Werror=format-security -Wno-format-nonliteral -Wshorten-64-to-32 -Wshadow -Werror=vla -Wdate-time -Wnull-dereference -Wextra-semi -Wcomma -Wfloat-conversion -Wdocumentation -Wno-documentation-deprecated-sync -Wunused-private-field -Wmissing-prototypes -Wmissing-declarations -Wnon-virtual-dtor -Woverloaded-virtual -fno-operator-names -Wzero-as-null-pointer-constant -Wsuggest-override -Wimplicit-fallthrough -O3 sdk/core/clinksegment.cpp -o ../o/clinksegment.o' failed (returned 1)
make[3]: *** [GNUmakefile:76: ../o/clinksegment.o] Error 1
make[3]: Leaving directory '/opt/gdaljs/gdal/frmts/pcidsk'
make[2]: *** [GNUmakefile:15: pcidsk-install-obj] Error 2
make[2]: Leaving directory '/opt/gdaljs/gdal/frmts'
make[1]: *** [GNUmakefile:103: frmts-target] Error 2
make[1]: Leaving directory '/opt/gdaljs/gdal'
emmake: error: 'make static-lib' failed (returned 2)
make: *** [Makefile:179: gdal/libgdal.a] Error 1
```

Any ideas on what might be going wrong here? [this](https://github.com/wxWidgets/wxWidgets/issues/18319) is a relevant thread.


On a side note, could verify the examples working on MacOS.

--- a/Dockerfile
+++ b/Dockerfile
@@ -22,8 +22,8 @@ WORKDIR /opt/emsdk
 # In that case, replace "latest" below with "${EMSCRIPTEN_V}".
 ARG EMSCRIPTEN_V
 
-RUN ./emsdk install latest
-RUN ./emsdk activate latest
+RUN ./emsdk install ${EMSCRIPTEN_V}
+RUN ./emsdk activate ${EMSCRIPTEN_V}
 
 RUN git config --global user.name 'Nobody'
 RUN git config --global user.email 'nobody@nowhere.nope'
@@ -35,5 +35,6 @@ COPY entrypoint.sh .
 RUN chmod +x entrypoint.sh
 
 WORKDIR /opt/gdaljs
+RUN git config --global --add safe.directory /opt/gdaljs
 
 ENTRYPOINT ["/opt/entrypoint.sh"]

--- a/Makefile
+++ b/Makefile
@@ -28,6 +28,8 @@ EXPORTED_FUNCTIONS = "[\
   '_GDALGetRasterMinimum',\
   '_GDALGetRasterMaximum',\
   '_GDALGetRasterNoDataValue',\
+  '_GDALRasterIO',\
+  '_GDALRasterIOEx',\
   '_GDALGetProjectionRef',\
   '_GDALSetProjection',\
   '_GDALGetGeoTransform',\
@@ -71,6 +73,8 @@ EXPORTED_FUNCTIONS = "[\
   '_GDALDatasetGetLayerByName',\
   '_GDALDatasetGetLayerCount',\
   '_GDALDatasetExecuteSQL',\
+  '_GDALDatasetRasterIO',\
+  '_GDALDatasetRasterIOEx',\
   '_OGR_L_GetNextFeature',\
   '_OGR_L_GetExtent',\
   '_OGR_L_GetLayerDefn',\

--- a/scripts/setup
+++ b/scripts/setup
@@ -4,5 +4,7 @@ set -e
 
 DIR="$(dirname "$0")"
 
+# Pin to emsdk 3.1.21. Higher versions seem to switch to C++17 by default, which GDAL 2.4 isn't
+# compatible with.
 docker build -t gdaljs-build "${DIR}/.." \
-    --build-arg EMSCRIPTEN_V=3.1.18
+    --build-arg EMSCRIPTEN_V=3.1.21

